src/Native/ObjWriter/objwriter.cpp

```diff
@@ -391,10 +396,12 @@ int ObjectWriter::EmitSymbolRef(const char *SymbolName,
   case RelocType::IMAGE_REL_BASED_REL32:
     Size = 4;
     IsPCRel = true;
+    Kind = MCSymbolRefExpr::VK_PLT;
```

Comment:
VK_PLT is for code symbols, and for data symbols it should be VK_GOTPCREL

Reply:
Thank you for looking into this! You probably know more about ELF relocs than me at this point.

Is there any way to express "this reloc points to something that is part of the ELF module and will for sure fit in the 32 bit reloc" because it's in the same file?

Part of the reason why I marked this pull request WIP is because I don't particularly like we have to go through jump stubs to access symbols that are defined within the same module (some of these relocs literally point to things that are in the same `.o` file, the others point to things that are in libRuntime.so). GOTPCREL will similarly require an indirection to access things that are right there.

If this is something that we have to live with, we can probably introduce a `--pic` command line option for ILC and have it generate the necessary indirections under that switch.

Reply:
I've just studied all this relocation tricks for ELF :)
But, as I understand - this indirections is a standard way of making shared libs in *nix.
There is question on StackOverflow about that: https://stackoverflow.com/questions/53086636/linking-a-static-library-into-a-shared-library-without-fpic

Reply:
I believe symbol type R_X86_64_PC32 means "32 bit offset relative to procedure call".
But LD will allow this type only for local symbols.
I've tried to make all symbols LOCAL, except ExternSymbolNode. Everything get linked, but still have a problem with stack unwinding.